src/exporters/clickhouse/transformer.rs

```diff
@@ -35,25 +35,69 @@ impl Transformer {
         }
     }
 
-    fn build_json_attrs(&self, attrs: &[ConvertedAttrKeyValue]) -> String {
+    fn build_json_attrs<'a>(
+        &self,
+        attrs: &'a [ConvertedAttrKeyValue],
+    ) -> HashMap<Cow<'a, str>, JsonType<'a>> {
         if attrs.is_empty() {
-            return "{}".to_string();
+            return HashMap::new();
         }
 
-        let hm: HashMap<Cow<'_, String>, String> = attrs
+        let hm: HashMap<Cow<'a, str>, JsonType<'a>> = attrs
             .iter()
             // periods(.) in key names will be converted into a nested format, so swap
             // them to underscores to avoid nesting
             .map(|kv| {
-                if self.use_json_underscore {
-                    (Cow::Owned(kv.0.replace(".", "_")), kv.1.to_string())
+                let key = if self.use_json_underscore {
+                    Cow::Owned(kv.0.replace(".", "_"))
                 } else {
-                    (Cow::Borrowed(&kv.0), kv.1.to_string())
-                }
+                    Cow::Borrowed(kv.0.as_str())
+                };
+                (key, (&kv.1).into())
+            })
+            .collect();
+
+        hm
+    }
+
+    pub(crate) fn transform_attrs_owned(
+        &self,
+        attrs: &[ConvertedAttrKeyValue],
+    ) -> MapOrJson<'static> {
+        match self.use_json {
+            true => MapOrJson::JsonOwned(self.build_json_attrs_owned(attrs)),
+            false => MapOrJson::Map(
+                attrs
+                    .iter()
+                    .map(|kv| (kv.0.clone(), kv.1.to_string()))
+                    .collect(),
+            ),
+        }
+    }
+
+    fn build_json_attrs_owned(
+        &self,
+        attrs: &[ConvertedAttrKeyValue],
+    ) -> HashMap<String, JsonType<'static>> {
+        if attrs.is_empty() {
+            return HashMap::new();
+        }
+
+        let hm: HashMap<String, JsonType<'static>> = attrs
```

Comment:
Noticed 'static here requires cloning, but sounds like we plan to remove in future iterations.